aptos-move/framework/src/natives/cryptography/algebra/serialization.rs

```diff
@@ -180,6 +185,161 @@ pub fn serialize_internal(
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FR_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            Ok(smallvec![Value::vector_u8(buf)])
+        },
+        (BN254Structure::BN254Fr, BN254SerializationFormat::BN254FrMsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fr, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FR_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            buf.reverse();
+            Ok(smallvec![Value::vector_u8(buf)])
+        },
+        (BN254Structure::BN254Fq, BN254SerializationFormat::BN254FrLsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fq, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FQ_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            Ok(smallvec![Value::vector_u8(buf)])
+        },
+        (BN254Structure::BN254Fq, BN254SerializationFormat::BN254FqMsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fq, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FQ_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            buf.reverse();
+            Ok(smallvec![Value::vector_u8(buf)])
+        },
+        (BN254Structure::BN254Fq2, BN254SerializationFormat::BN254Fq2LscLsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fq2, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FQ2_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            buf.reverse();
+            Ok(smallvec![Value::vector_u8(buf)])
+        },
+        (BN254Structure::BN254Fq12, BN254SerializationFormat::BN254Fq12LscLsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fq12, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FQ12_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            buf.reverse();
```

Comment:
unlike BSL12381Fq12LscLsb, here a reverse is needed? Let me double check.

Reply:
Oh, nice catch! that's a bug! I may wrong copy pasted from msb part.

Reply:
fixed now.